fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/election/fragments/ElectionFragment.java

```diff
@@ -150,36 +181,42 @@ public View onCreateView(
                   "User should not be able to use the action button in this state :" + state);
           }
         });
-    viewModel
-        .getCurrentLao()
-        .observe(
-            getViewLifecycleOwner(),
-            laoView -> {
-              Log.d(TAG, "lao updated");
-              setupElectionContent();
-            });
 
     return view;
   }
 
+  @Override
+  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+
+    try {
+      disposables.add(
+          electionRepository
+              .getElectionObservable(viewModel.getLaoId(), electionId)
+              .subscribe(
```

Comment:
Also, is the subscription guaranteed to be executed on the main thread ? If not it will be a problem. I think that `observeOn(AndroidSchedulers.mainThread())` guarantees it.